docs: change linting CLI and add custom rules

aepfli opened the pull request:
Changing to Markdownlint-cli2 and adding custom rules for better
usage.

closes: #974

aepfli commented:
no clue why unit and integration tests are failing

mowies commented on docs/Makefile:
```diff
@@ -15,7 +15,7 @@ shell:
 
 server:
-	$(DOCKER_CMD) $(VOLUMES) -p  $(PORT):$(PORT) $(IMAGE) server -D -p $(PORT)
+	$(DOCKER_CMD) $(VOLUMES) -p  $(PORT):$(PORT) $(IMAGE) server -D -p $(PORT) --disableFastRender
```
why did you add this flag?

aepfli replied:
```suggestion
	$(DOCKER_CMD) $(VOLUMES) -p  $(PORT):$(PORT) $(IMAGE) server -D -p $(PORT)
```

aepfli commented on docs/Makefile:
```diff
@@ -35,14 +35,15 @@ lint: lint-fix
 # - .markdownlintignore holds the configuration for files to be ignored
 # - .markdownlint.yaml contains the rules for markdownfiles
-#
+#davidanson/markdownlint-cli2-rules:next
```
```suggestion
```
copy paste error, will be deleted

aepfli commented on docs/content/en/docs/snippets/tasks/install.md:
```diff
@@ -51,8 +51,8 @@ kubectl wait --for=condition=Available deployment/lifecycle-operator -n keptn-li
-> **Note:**
-Installation of the Lifecycle Toolkit version 0.6.0 and lower is not supported via helm charts.
+> **Note**
+Installation of the Lifecycle Toolkit version 0.5.0 and lower is not supported via helm charts.
```
```suggestion
Installation of the Lifecycle Toolkit version 0.6.0 and lower is not supported via helm charts.
```

aepfli commented on docs/rules/README.md:
```diff
@@ -0,0 +1,25 @@
+Therefore we copied the rule into our Git Repository.
+This way they are working within IDEs and our makefile docker images.
+
+## Rules
```
created #1048

StackScribe commented on .markdownlint-cli2.yaml:
```diff
@@ -0,0 +1,20 @@
+config:
+  line-length:
+    line_length: 120
```
Is there an exception for code blocks and xref lines?

aepfli replied:
normally urls should not affect it - but i only migrated this from the old to the new file

StackScribe commented on CONTRIBUTING.md:
```diff
@@ -211,7 +212,8 @@ git push --set-upstream origin feature/123/foo
 
-Make sure you **sign off your commits**. To do this automatically check [this](https://github.com/keptn/lifecycle-toolkit/blob/main/CONTRIBUTING.md#auto-signoff-commit-messages).
+Make sure you **sign off your commits**.
+To do this automatically check [this](https://github.com/keptn/lifecycle-toolkit/blob/main/CONTRIBUTING.md#auto-signoff-commit-messages).
```
Using "this" or "here" in cases like this is not a realy good practice.  Better to use the title of the document being referenced.

aepfli replied:
i did not provide this content, i only added a break for better maintainability

StackScribe commented on README.md:
```diff
@@ -117,7 +119,8 @@ app.kubernetes.io/name: myAwesomeWorkload
 
-In general, the Keptn Annotations/Labels take precedence over the Kubernetes recommended labels. If there is no version
+In general, the Keptn Annotations/Labels take precedence over the Kubernetes recommended labels.
+If there is no version
 annotation/label and there is only one container in the pod, the Lifecycle Toolkit will take the image tag as version (
```
```suggestion
If no version annotation/label exists
and the pod only contains one container, the Lifecycle Toolkit takes the image tag as version
```

StackScribe reviewed (dismissed):
I left a couple suggestions and questions for your consideration but this looks good.  I made one suggestion against a file that is only here to fix markdown practices and then realized that these files should be edited in separate PRs so I didn't do any more.

mowies commented on CONTRIBUTING.md:
```diff
@@ -112,7 +112,11 @@ of the settings.
 
 If you are on Windows, you need to install **make** for the above process to complete.
 
+<<<<<<< HEAD
 > **Note**
+=======
+> **Note**:
+>>>>>>> 1c24fa0 (fixing issues)
```
you have some merge artifacts

aepfli replied:
i hate this pr by now - i already rebased it so often - because of merge conflicts :( - and always is there something more

aepfli replied:
deleted my fix commit, and reapplied markdownlint and created one new commit for each rule.

mowies commented on docs/Makefile:
```diff
@@ -35,14 +35,14 @@ lint: lint-fix
 markdownlint:
-	$(MDL_CMD)
+	$(MDL_CMD) davidanson/markdownlint-cli2-rules:next "**/*.md" 
 
 markdownlint-fix:
-	$(MDL_CMD) --fix
+	$(MDL_CMD) --entrypoint="markdownlint-cli2-fix" davidanson/markdownlint-cli2-rules:next "**/*.md" 
```
can we use pinned version tags for the image here?

aepfli replied:
since we opened this pr there was finally the official release, and we do not need to use next anymore, will adapt it

aepfli replied:
i will drop my commit with the applied changes - and add a new one - this should make things easier

aepfli replied:
markdownlint-cli2 is still not release https://hub.docker.com/r/davidanson/markdownlint-cli2-rules/tags - i added a comment also for the renovate update :) we need to delete a line with the next version

mowies commented on docs/content/en/docs/snippets/tasks/install.md:
```diff
@@ -57,4 +57,21 @@ kubectl wait --for=condition=Available deployment/lifecycle-operator -n keptn-li
 The Lifecycle Toolkit and its dependencies are now installed and ready to use.
 
-</details>
+> **Note**
```
what happened here? did you intentionally change the content?

aepfli replied:
https://github.com/keptn/lifecycle-toolkit/pull/1031#discussion_r1156777688

mowies commented on mdl.log:
```diff
@@ -0,0 +1,7 @@
+docs/content/en/_index.md:28:77 MD034/no-bare-urls Bare URL used [Context: "https://github.com/keptn/lifec..."]
```
this was probably pushed by accident

aepfli replied:
deleted

mowies commented:
@aepfli ping when you need another review pls

aepfli commented:
To make life easier for all:
- the first [4 commits](https://github.com/keptn/lifecycle-toolkit/pull/1031/files/8e84b3da91b14e6719ef8d63106500cd62f73d67) contain the changes regarding migration and additional rules
- [admonition fixes](https://github.com/keptn/lifecycle-toolkit/pull/1031/commits/d44492de17db8b31c5fc14762d4c2199e8492ba6)
- [max 1 sentence per line fixes](https://github.com/keptn/lifecycle-toolkit/pull/1031/commits/abe6b5281519d0a4daff765ba076ae5ca21533d2)

I changed the git history a bit and removed the commit, in which I applied the changes and readded those changes at the end. This removes all the errors due to regular rebasing and other conflicts. I'll do the same thing again if I have merge conflicts, as the first four commits are important ones.

RealAnna commented on README.md:
```diff
@@ -65,14 +65,16 @@ kubectl apply -f https://github.com/keptn/lifecycle-toolkit/releases/download/v0
 is handled automatically
-by [klt-cert-manager](https://github.com/keptn/lifecycle-toolkit/blob/main/klt-cert-manager/README.md). Versions 0.5.0
+by [klt-cert-manager](https://github.com/keptn/lifecycle-toolkit/blob/main/klt-cert-manager/README.md).
+Versions 0.5.0
```
this is actually valid for 0.6.0 since we postponed the release of the helm chart, would you mind correcting it ?

RealAnna replied:
```suggestion
Versions 0.6.0
```
